k8s_dp/k8s_dp.p4

```diff
@@ -501,6 +574,247 @@ control k8s_dp_control(
+            meta.range_check_result[5:5] = (bit<1>)((min5 <= hdr.tcp.dst_port) && (hdr.tcp.dst_port <= max5));
+            meta.range_check_result[6:6] = (bit<1>)((min6 <= hdr.tcp.dst_port) && (hdr.tcp.dst_port <= max6));
+            meta.range_check_result[7:7] = (bit<1>)((min7 <= hdr.tcp.dst_port) && (hdr.tcp.dst_port <= max7));
+        }
+
+        action do_range_check_udp(
+            bit<16> min0, bit<16> max0,
+            bit<16> min1, bit<16> max1,
+            bit<16> min2, bit<16> max2,
+            bit<16> min3, bit<16> max3,
+            bit<16> min4, bit<16> max4,
+            bit<16> min5, bit<16> max5,
+            bit<16> min6, bit<16> max6,
+            bit<16> min7, bit<16> max7)
+        {
+            meta.range_check_result[0:0] = (bit<1>)((min0 <= hdr.udp.dst_port) && (hdr.udp.dst_port <= max0));
+            meta.range_check_result[1:1] = (bit<1>)((min1 <= hdr.udp.dst_port) && (hdr.udp.dst_port <= max1));
+            meta.range_check_result[2:2] = (bit<1>)((min2 <= hdr.udp.dst_port) && (hdr.udp.dst_port <= max2));
+            meta.range_check_result[3:3] = (bit<1>)((min3 <= hdr.udp.dst_port) && (hdr.udp.dst_port <= max3));
+            meta.range_check_result[4:4] = (bit<1>)((min4 <= hdr.udp.dst_port) && (hdr.udp.dst_port <= max4));
+            meta.range_check_result[5:5] = (bit<1>)((min5 <= hdr.udp.dst_port) && (hdr.udp.dst_port <= max5));
+            meta.range_check_result[6:6] = (bit<1>)((min6 <= hdr.udp.dst_port) && (hdr.udp.dst_port <= max6));
+            meta.range_check_result[7:7] = (bit<1>)((min7 <= hdr.udp.dst_port) && (hdr.udp.dst_port <= max7));
+        }
+
+        action do_range_check_sctp(
+            bit<16> min0, bit<16> max0,
+            bit<16> min1, bit<16> max1,
+            bit<16> min2, bit<16> max2,
+            bit<16> min3, bit<16> max3,
+            bit<16> min4, bit<16> max4,
+            bit<16> min5, bit<16> max5,
+            bit<16> min6, bit<16> max6,
+            bit<16> min7, bit<16> max7)
+        {
+            meta.range_check_result[0:0] = (bit<1>)((min0 <= hdr.sctp.dst_port) && (hdr.sctp.dst_port <= max0));
+            meta.range_check_result[1:1] = (bit<1>)((min1 <= hdr.sctp.dst_port) && (hdr.sctp.dst_port <= max1));
+            meta.range_check_result[2:2] = (bit<1>)((min2 <= hdr.sctp.dst_port) && (hdr.sctp.dst_port <= max2));
+            meta.range_check_result[3:3] = (bit<1>)((min3 <= hdr.sctp.dst_port) && (hdr.sctp.dst_port <= max3));
+            meta.range_check_result[4:4] = (bit<1>)((min4 <= hdr.sctp.dst_port) && (hdr.sctp.dst_port <= max4));
+            meta.range_check_result[5:5] = (bit<1>)((min5 <= hdr.sctp.dst_port) && (hdr.sctp.dst_port <= max5));
+            meta.range_check_result[6:6] = (bit<1>)((min6 <= hdr.sctp.dst_port) && (hdr.sctp.dst_port <= max6));
+            meta.range_check_result[7:7] = (bit<1>)((min7 <= hdr.sctp.dst_port) && (hdr.sctp.dst_port <= max7));
+        }
+
+        action do_range_check_udplite(
```

Comment:
Do we need to support UDP lite? It's only for IOT with slightly damaged payloads.

Reply:
Yes, we do. Both UDPLite and SCTP would be added subsequently. For now, we are limiting to only TCP and UDP.

Reply:
I am assuming, we will match the fields on DPDK side as on MEV side. It may not affect API but it affects store to p4runtime which again is a common code.

Reply:
UPDlite is also used in 5G. We had explicit requirement from E/// for that in my previous project.
And yes, we will match all such things between MEV with DPDK unless there is a difference due to architectural differences between the two targets.